### PR TITLE
Semester wise grades

### PR DESCRIPTION
**Pull request by kishore-narendran**

### CODE REVIEW

**Comment by aneesh-neelam** [comment]
Hey @kishore-narendran, just get Semester-wise GPA and do not split it into multiple course lists.

File: api/scraper/grades.js [hunk 1]
```diff
@@ -55,13 +56,35 @@ exports.get = function (app, data, callback) {
...
+              examHeldCollection.shift();
+              courses.shift();
+              var semesterWiseGrades = {};
+
+              //Consolidating the grades in a semester wise fashion
+              for(var i = 0; i < examHeldCollection.length; i++) {
+                var examHeld = examHeldCollection[i];
+                semesterWiseGrades[examHeld] = [];
+              }
+
```
Comment (aneesh-neelam) [inline api/scraper/grades.js:142]: No need of this, just put in an object that has exam_held as keys and GPAs as values.

**Comment by kishore-narendran** [comment]
You want me to get you only a GPA? Not the course listings?

**Comment by aneesh-neelam** [comment]
Yes. Semester-wise GPA only. Because everything else is already there.

**Comment by kishore-narendran** [comment]
Okay, I thought you wanted the semester wise courses too for accessibility.

**Comment by kishore-narendran** [comment]
@aneesh-neelam : Have a look now

File: api/scraper/grades.js [hunk 2]
```diff
@@ -55,13 +84,46 @@ exports.get = function (app, data, callback) {
...
+                //Aggregating the unique semesters by checking and adding the unique EXAM_HELD values
+                var examHeld = moment(attrs.eq(6).text()).format('YYYY-MM');
+                if(!(examHeldCollection.indexOf(examHeld) > -1)) {
+                  examHeldCollection.push(examHeld);
                 }
+                semesterWiseGrades[examHeld] = [];
+                courses.push(course);
               });
 
+              //Removing the column headers from the data
```
Comment (aneesh-neelam) [inline api/scraper/grades.js]: This is unnecessary, check the [latest development branch on upstream](https://github.com/aneesh-neelam/VITacademics/tree/development).

**Comment by aneesh-neelam** [comment]
@kishore-narendran: Check the [latest development branch on upstream](https://github.com/aneesh-neelam/VITacademics/tree/development). Make the changes accordingly, it'll be easier to merge then.

**Comment by karthikb351** [comment]
Why is @kishore-narendran sending PR's from his fork? Make new branches on this repo only and send a PR. That way all of us can amend to it if necessary.

**Comment by aneesh-neelam** [comment]
I asked him to do it that way only. Let it be.

**Pull request closed without merging**